flax/core/lift.py

```diff
@@ -177,7 +177,7 @@ def repack(inner_scope_tree):
       y, out_variable_groups_xs_t = fn(
           scope_fn, repack,
           variable_groups_xs_t, rng_groups_xs_t,
-          *args)
+          *args, **kwargs)
```

Comment:
We would need to handle kwargs in other transformations as well. This might be tricky as JAX transforms don't always handle kwargs either.

I think it's also better to pass kwargs as a dict internally to avoid collisions with arg names.

Reply:
Does this sounds like a parameter named kwargs but only unpacked later at [here](https://github.com/google/flax/blob/3cea8ad2ad49ec53ae7ad3eefded1ba5b1f557d5/flax/core/lift.py#L671)? I'm not quite clear about the JAX part.

Reply:
I think for remat JAX supports it and for jit support was added last week I think. This needs to be checked per transform though. For example with jit you also now need static_arg_names to pass kwargs statically (which is usually the case)

Reply:
In this case, what about adding a judgement of the `name` parameter at the beginning of `pack`? It's like when `name` is jit and the function's co_kwonlyargcount is not zero, an error(unimplemented) will be raised.

Reply:
I think it would be cleaner to add a kwarg to pack indicating whether kwargs is supported instead of using string matching to figure out who called the function.

Reply:
What about adding a parameter like enable_kwargs with default value False, and then set it True in the `lift.checkpoint`?

Reply:
Yeah I agree the default should be False